app/controller/grid/ItemDeleterGridControllerMixin.js

```diff
@@ -1,44 +1,59 @@
+
+/**
+ *  To be mixed in to the controller of a grid which uses ItemDeleter
+ *  Can be provided with the following custom handlers 
+ * 
+ *  - beforeDelete          => if returns "false" deletion is aborted, runs BEFORE the user is asked for confirmation
+ * 
+ *  - onRowDeleteFail       => do something if the delete failed
+ *  - onRowDeleteSuccess    => do something if the delete succeeded
+ *  - onRowDeleteCallback   => do something whether the delete succeeded or failed
+ * 
+ *  - doDelete              => the actual deletion implementation, if the user confirms
+ *  - dontDelete            => in case we want to do something when the user doesn't confirim or beforeDelete returns false
+ */
+
 Ext.define('CpsiMapview.controller.grid.ItemDeleterGridControllerMixin', {
     extend: 'Ext.Mixin',
 
+    beforeDelete: function(config) {return true;},
+    onRowDeleteFail: Ext.emptyFn,
+    onRowDeleteSuccess: Ext.emptyFn,
+    onRowDeleteCallback: Ext.emptyFn,
+
+    // what to do for deleting
+    doDelete: function (record) {
+        record.erase({
+            failure: this.onRowDeleteFail,
+            success: this.onRowDeleteSuccess,
+            callback: this.onRowDeleteCallback
+        });
+    },
+
+    // what to do for not deleting
+    dontDelete: Ext.emptyFn,
+
     onDeleteRowClick: function (tableView, rowIndex, colIndex, item, e, record, tableRow) {
         var me = this;
-        // what to do for deleting
-        var doDelete = function () {
-            record.erase({
-                failure: function (/*record, operation*/) {
-                    // do something if the delete failed
-                },
-                success: function (/*record, operation*/) {
-                    // do something if the delete succeeded
-                },
-                callback: function (/*record, operation, success*/) {
-                    // do something whether the delete succeeded or failed
-                }
-            });
-        };
-        // what to do for not deleting
-        var dontDelete = function () {
-        };
-        var beforeDeleteConfig = { avoidConfirmationRequest: item.avoidConfirmationRequest || false };
+        var beforeDeleteConfig = { avoidConfirmationRequest: item.avoidConfirmationRequest || false, record: record };
```

Comment:
Is item.avoidConfirmationRequest an ExtJS property? If not where is this set?

Reply:
It's on the ItemDeleter component